backend/tilavarauspalvelu/integrations/keyless_entry/service.py

```diff
@@ -413,90 +410,169 @@ def sync_access_code(cls, obj: ApplicationSection | RecurringReservation | Reser
-                ),
-                recurring_reservation=None,
-                access_code_generated_at__isnull=False,
-                end__gt=local_datetime(),
-            )
+    def _create_missing_access_codes_for_series(cls) -> None:
+        """
+        Create access codes for recurring reservations that are missing them.
+        Do not include series in seasonal bookings.
+        """
+        all_series: Iterable[RecurringReservation] = RecurringReservation.objects.requiring_access_code().filter(
+            allocated_time_slot__isnull=True
         )
 
-        if not reservations:
-            return
+        for series in all_series:
+            is_active: bool = series.should_have_active_access_code  # type: ignore[attr-defined]
 
-        for reservation in reservations:
             with suppress(ExternalServiceError):
-                if reservation.access_code_should_be_active:
-                    try:
-                        PindoraClient.activate_reservation_access_code(reservation=reservation)
+                try:
+                    cls.create_access_code(obj=series, is_active=is_active)
 
-                    # If we think an access code has been generated, but it's not found in Pindora,
-                    # set `access_code_generated_at`. `create_missing_pindora_reservations` will
-                    # then create the access code if needed.
-                    except PindoraNotFoundError:
-                        reservation.access_code_generated_at = None
+                # If series already exists, fetch it and update instead.
+                except PindoraConflictError:
+                    response = cls.get_access_code(obj=series)
 
-                    else:
-                        reservation.access_code_is_active = True
+                    series.reservations.requiring_access_code().update(
+                        access_code_generated_at=response.access_code_generated_at,
+                        access_code_is_active=response.access_code_is_active,
+                    )
+
+    @classmethod
+    def _create_missing_access_codes_for_seasonal_bookings(cls) -> None:
+        """Create access codes for application sections that are missing them."""
+        sections: Iterable[ApplicationSection] = ApplicationSection.objects.requiring_access_code()
+
+        for section in sections:
+            is_active: bool = section.should_have_active_access_code  # type: ignore[attr-defined]
+
+            with suppress(ExternalServiceError):
+                try:
+                    cls.create_access_code(obj=section, is_active=is_active)
+
+                # If seasonal booking already exists, fetch it and update instead.
+                except PindoraConflictError:
+                    response = cls.get_access_code(obj=section)
+
+                    section.actions.get_reservations().requiring_access_code().update(
```

Comment:
`get_reservations().update_access_code_info` would also be safe to call here?

Reply:
Calling reschedule now, with additional activate/deactivate calls if that state seems incorrect.